modules/ROOT/pages/sources-reference.adoc

```diff
@@ -0,0 +1,69 @@
+= Mule Sources
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:keywords: anypoint studio, flow, source, sources, redelivery, policy
+
+Sources are a very important part of any Mule Application. When added to a flow, they serve as the entry point for information,
+linking any external communication protocol with internal Mule processing.
+
+Examples of sources are:
+
+* HTTP:Listener
+* FILE:Inbound-endpoint
+* AMQP:Consume
+
+== Configuring Sources
+
+Even though each source implementation will have different configuration parameters, there are some common ones worth mentioning.
+
+=== Redelivery Policy
+
+Each source can be configured with a redelivery policy. That way, an user can add some logic on how to handle messages that arrive at the source multiple times and cause exceptions.
+
+For example:
+[source,xml]
+----
+<http:listener path="exhaustRedelivery" config-ref="listenerConfig">
+    <redelivery-policy maxRedeliveryCount="${maxRedeliveryCount}" idExpression="#[vars.idExpression]"/>
+</http:listener>
+----
+
+==== Redelivery Policy Configuration Parameters
+
+[%header,cols="20a,20a,20a,60a"]
+|===
+|Field |Type |Default Value |Description
+|*Max Redelivery Count* |Number |5 |The maximum number of times a message can be redelivered and processed unsuccessfully before triggering process-failed-message
+|*Use Secure Hash* |Boolean |True |Whether to use a secure hash algorithm to identify a redelivered message.
+|*Message Digest Algorithm* |String |SHA-256 |The secure hashing algorithm to use.
+|*ID Expression* |Expression|- | Defines one or more expressions to use to determine when a message has been redelivered.This property may only be set if useSecureHash is false.
+|*Object Store* |String |An Object Store created by the ObjectStoreManager. Non-persistent, with an entryTtl of 300 SECONDS and an expiration interval of 6 SECONDS |The object store where the redelivery counter for each message is going to be stored. It can be a reference of a private-object-store can be created as child element.
```

Comment:
`private-object-store`? is that correct or a regular os can be used?

Reply:
Any of those. The attribute can reference a globally created object store by name or a private-object-store child element can be created.

Reply:
I would not specify anything then, and just mention you can configure it as a reference or an inner element, linking to the OS docs.